### ISSUE TITLE
Streamline graph output (graphviz, mermaid, etc.)

### ISSUE THREAD

**Issue by abejenaru** [tox-dev/pipdeptree#349]
### Describe the feature

Hi, first of all big thanks for creating pipdeptree :)

I was looking into adding support for D2 graph generation, and I think it would be better to avoid adding another output option at this time.
Maybe streamlining the current graphviz and mermaid options under a common option (for example --output mermaid, --output graphviz_dot, etc.) and then supporting new formats is a workable enhancement ?

**Comment by kemzeb** [comment]
Though the D2 project does look fairly new (compared to mermaid and graphviz), I personally wouldn't mind this being added!

> Maybe streamlining the current graphviz and mermaid options under a common option (for example --output mermaid, --output graphviz_dot, etc.) and then supporting new formats is a workable enhancement ?

I think this is a great idea too (though we would need to keep the older options for compatibility). We can have something like the following:
```
$ pipdeptree -h
usage: pipdeptree . . .
. . .
--output       choose how you want to render; this includes mermaid, graphviz-dot, graphviz-png, graphviz-svg, ... (default: text)
--mermaid  (DEPRECATED, use --option instead) . . .
. . .

**Comment by abejenaru** [comment]
PS. Would adding --json and --json-tree rendering options in there make sense ? For me they are at the same level as graph outputs, but what is your take ?
Also, personal preference would be for a 1-letter parameter, like `-o --output`, as `-r` is already in use.

**Comment by kemzeb** [comment]
> PS. Would adding --json and --json-tree rendering options in there make sense ? For me they are at the same level as graph outputs, but what is your take ?

Yes, since both --json and --json-tree are rendering options we should include them. To me it would be confusing to see them as their own options but throw the rest within --output.

> Also, personal preference would be for a 1-letter parameter, like -o --output, as -r is already in use.

Sure, we can do this instead.

**Comment by kemzeb** [comment]
Just want to note here, for the future, that graphviz supports many different types of output formats. It doesn't make sense to to add more and more values for it e.g. `graphviz-json`, `graphviz-png`, `graphviz-svg`, ....).

A better approach would be to:
- Look for a string that starts with `graphviz-` when command-line parsing `--output`
- Capture everything after the `-` into its own variable
- Pass that variable to the graphviz render function like how we would normally do
  - The render code, as it already does, will validate that this is a valid format that graphviz supports
